src/middlewared/middlewared/plugins/webdav.py

```diff
@@ -24,6 +25,8 @@ class Config:
             )
         else:
             await check_path_resides_within_volume(verrors, self.middleware, f'{schema}.path', data['path'])
+            if not verrors and not os.path.exists(path):
```

Comment:
There could be errors after this and the directory will be create.

It seems to make more sense to create only after, if there are no errors.